### ISSUE TITLE
Suggestion: Add Agricraft compatability for the new Harvesting Drone

### ISSUE THREAD

**Issue by wrincewind** [TeamPneumatic/pnc-repressurized#211]
Currently the drone just sits there, floating next to my fully grown crops like some big fat idiot. >:I 

[TBF I have a nice little program for doing this with proper drones, but... it'd be nice, right?

**Comment by desht** [comment]
Yeah, I think this shouldn't be too hard.  Looks like Agricraft crops are just a tile entity with some handy methods for checking maturity and doing harvesting. @MineMaarten wrote a nice extensible system for crop harvesting so plugging in Agricraft should be fairly straightforward.

**Comment by wrincewind** [comment]
Brilliant, thanks! Now I just need to pass this onto EnderIO, thaumcraft, and Industrial foregoing, 'cause none of them seem to know how to do it either :p

Sent from Yahoo Mail on Android 
 
  On Tue, 10 Jul 2018 at 10:40 am, desht<notifications@github.com> wrote:   
Yeah, I think this shouldn't be too hard. Looks like Agricraft crops are just a tile entity with some handy methods for checking maturity and doing harvesting. @MineMaarten wrote a nice extensible system for crop harvesting so plugging in Agricraft should be fairly straightforward.

**Comment by desht** [comment]
Actually, no - I was wrong.  The example I was looking at (Botania Tweaks) uses ASM to patch Agricraft to make the crops harvestable.  I'm not touching ASM with a bargepole...

**Comment by wrincewind** [comment]
Blast, I had a feeling it couldn't be that easy... I'll go and pester the Agricraft devs to see if they can add something. 

Sent from Yahoo Mail on Android 
 
  On Tue, 10 Jul 2018 at 11:01 am, desht<notifications@github.com> wrote:   
Actually, no - I was wrong. The example I was looking at (Botania Tweaks) uses ASM to patch Agricraft to make the crops harvestable. I'm not touching ASM with a bargepole...

**Comment by desht** [comment]
I'll continue to research.  I'm pretty sure there's a way...

**Comment by Sewef** [comment]
Back to 2017, with Agricraft 1.10.2 (alpha build), I went into compatibility issues : machines from others mods couldn't harvest crops without using a FakePlayer (Mechanical User from Ex2, Drone from PNC, Turtle from CC, Robot from OC)... I checked the Agricraft Github and they just removed all mod compat with 1.10 update and they were going to re-add them.
As Agricraft 1.12.2 is still in alpha, it has the same non-compat problem, I guess. That's one the the reasons that made me uninstall this mod :l

**Comment by wrincewind** [comment]
Buggery. Maybe I should swap it for magical crops or something similar after all, then... Its that or I start handing out custom-made farmer drones like frigging candy :I

Sent from Yahoo Mail on Android 
 
  On Tue, 10 Jul 2018 at 1:35 pm, Sewef<notifications@github.com> wrote:   
Back to 2017, with Agricraft 1.10.2 (alpha build), I went into compatibility issues : machines from others mods couldn't harvest crops without using a FakePlayer (Mechanical User from Ex2, Drone from PNC, Turtle from CC, Robot from OC)... I checked the Agricraft Github and they just removed all mod compat with 1.10 update and they were going to re-add them.
As Agricraft 1.12.2 is still in alpha, it has the same non-compat problem, I guess. That's one the the reasons that made me uninstall this mod :l

**Comment by desht** [comment]
I'm looking at https://github.com/AgriCraft/AgriCraft/tree/master/src/main/java/com/infinityraider/agricraft/api/v1 now, and it does appear to have the necessary support for checking if crops are mature and/or harvestable.  It *looks* workable, but I won't know until I've tried some integration...

**Comment by wrincewind** [comment]
Well, fingers crossed...!
